pkg/distributor/rate_strategy.go

```diff
@@ -0,0 +1,109 @@
+// SPDX-License-Identifier: AGPL-3.0-only
+// Provenance-includes-location: https://github.com/cortexproject/cortex/blob/master/pkg/distributor/ingestion_rate_strategy.go
+// Provenance-includes-license: Apache-2.0
+// Provenance-includes-copyright: The Cortex Authors.
+
+package distributor
+
+import (
+	"math"
+
+	"github.com/grafana/dskit/limiter"
+	"golang.org/x/time/rate"
+
+	"github.com/grafana/mimir/pkg/util/validation"
+)
+
+// ReadLifecycler represents the read interface to the lifecycler.
+type ReadLifecycler interface {
+	HealthyInstancesCount() int
+}
+
+type globalStrategy struct {
+	baseStrategy limiter.RateLimiterStrategy
+	ring         ReadLifecycler
+}
+
+func newGlobalRateStrategy(baseStrategy limiter.RateLimiterStrategy, ring ReadLifecycler) limiter.RateLimiterStrategy {
+	return &globalStrategy{
+		baseStrategy: baseStrategy,
+		ring:         ring,
+	}
+}
+
+func (s *globalStrategy) Limit(tenantID string) float64 {
+	numDistributors := s.ring.HealthyInstancesCount()
+
+	if numDistributors == 0 {
+		return s.baseStrategy.Limit(tenantID)
+	}
+
+	return s.baseStrategy.Limit(tenantID) / float64(numDistributors)
```

Comment:
I think that if `s.baseStrategy.Limit(tenantID)` is `rate.Inf` then we should just return it as is.